src/StatusTimer.cpp

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#include "StatusTimer.h"
+#include "json.h"
+
+namespace Forwarder {
+
+void StatusTimer::start() {
+  Logger->trace("Starting the StatusTimer");
+  Running = true;
+  AsioTimer.async_wait(
+      [this](std::error_code const & /*error*/) { this->reportStatus(); });
+  StatusThread = std::thread(&StatusTimer::run, this);
+}
+
+void StatusTimer::waitForStop() {
+  Logger->trace("Stopping StatusTimer");
+  Running = false;
```

Comment:
The `Running` member and `AsioTimer.cancel();` can be removed. Simply call `IO.stop();`.